### PR TITLE
[HIP] Add support for querying USM allocation size

### PR DESCRIPTION
**Pull request by frasercrmck**

### CODE REVIEW

File: source/adapters/hip/usm.cpp [hunk 1]
```diff
@@ -230,8 +230,12 @@ urUSMGetMemAllocInfo(ur_context_handle_t hContext, const void *pMem,
       }
       return ReturnValue(Pool);
     }
+    case UR_USM_ALLOC_INFO_SIZE: {
+      size_t RangeSize;
```
Comment (jchlanda) [inline source/adapters/hip/usm.cpp:234]: Not sure if `pedantic` would complain about uninitialised variable here (probably yes), but you might want to give `RangeSize` a default value. Like we do in [here](https://github.com/oneapi-src/unified-runtime/blob/0959c61a557127baeec9cdcdc807969d647d84b9/source/adapters/hip/device.cpp#L167) for example.

File: source/adapters/hip/usm.cpp [hunk 2]
```diff
@@ -230,8 +230,12 @@ urUSMGetMemAllocInfo(ur_context_handle_t hContext, const void *pMem,
...
+      UR_CHECK_ERROR(hipMemPtrGetInfo(const_cast<void *>(pMem), &RangeSize));
```
Comment (jchlanda) [inline source/adapters/hip/usm.cpp:235]: :)

**Review by jchlanda — APPROVED** [review]

File: source/adapters/hip/usm.cpp (hunk 1 above)
Comment (frasercrmck) [reply source/adapters/hip/usm.cpp:234]: Good shout!

File: source/adapters/hip/usm.cpp (hunk 2 above)
Comment (frasercrmck) [reply source/adapters/hip/usm.cpp:235]: Yeah.. I don't feel too bad as it's done in quite a few other places.

Comment (jchlanda) [reply source/adapters/hip/usm.cpp:235]: It's on HIP, great API design.

**Pull request merged**